### PR TITLE
GH-37051: [Dev][JS] Add Dependabot configuration for npm

### LINKED ISSUE #37051: [Dev][JS] Add Dependabot configuration for npm
### Describe the enhancement requested

Target PR examples:

* https://github.com/apache/arrow/pull/36603
* https://github.com/apache/arrow/pull/36761

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#yarn

### Component(s)

Developer Tools, JavaScript

### PR DESCRIPTION
**Pull request by kou**
### Rationale for this change

We can add `MINOR: [JS] ` prefix to PRs from Dependabot automatically.

### What changes are included in this PR?

Add a configuration for npm.

### Are these changes tested?

No. I want to test this by merging this to main.

### Are there any user-facing changes?

No.
* Closes: #37051

### CODE REVIEW

**Review by assignUser — APPROVED** [review]
Sounds good. @domoritz what do you think?

**Comment by domoritz** [comment]
I don't think we need weekly updates since it would just add extra work to review and merge.

Do we have a script to automatically merge patch and minor updates that don't break the tests?

**Comment by assignUser** [comment]
> I don't think we need weekly updates since it would just add extra work to review and merge.

We can change the interval to monthly?

> Do we have a script to automatically merge patch and minor updates that don't break the tests?

While technically possible I don't think the ASF policy would allow that (only if the bot was using a PAT by a committer who would then be personally responsible for any changes contributed with their PAT.)

**Comment by kou** [comment]
If we can merge PRs from Dependabot only when CI is green, I'll do it. It's not a large work.

> Do we have a script to automatically merge patch and minor updates that don't break the tests?

We can do it by a `on: pull_request_target` GitHub Actions workflow. But it's out-of-scope of this. This just wants to prepare title of PRs from Dependabot. Formatting PR titles by hand is a very bother task for me.

**Comment by domoritz** [comment]
That all sounds good to me. I'd change it to monthly.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@ updates:
       interval: "weekly"
     commit-message:
       prefix: "MINOR: [CI] "
+  - package-ecosystem: "npm"
+    directory: "/js/"
+    schedule:
+      interval: "weekly"
```
Comment (kou) [inline .github/dependabot.yml:29]:
```suggestion
      interval: "monthly"
```

**Comment by kou** [comment]
OK. I've changed to "monthly".
I'll merge this.

**Pull request merged**